### ISSUE TITLE
Optional unwrapping failure OidcIdentityProvider at l. 223

### ISSUE THREAD

**Issue by artkonr** [quarkusio/quarkus#21146]
### Describe the bug

Hey team,

Recently encountered a weird bug in OIDC: a redirect loop is started because of an authentication failure on the stage of accepting the `code` from the IDP. 

Superficial debugging, has revealed that the loop occurs because of consistently thrown `AuthenticationFailedException`, which in turn is caused by internally-thrown `NoSuchElementException`. There seems to be a `Optional` unwrapping failure at line 223 in `OidcIdentityProvider`. I see that the method is `@Deprecated`, but it seems to be causing a bug anyway.

Supposedly, the problem has started when I added these 2 properties:
```text
quarkus.oidc.token.refresh-expired=true
quarkus.oidc.token.refresh-token-time-skew=5S
```

In the logs it's shown only as "Authentication failure: java.util.NoSuchElementException: No value present". Even if the described behavior is intended, the message looks cryptic, although it seems to be notifying of a misconfiguration (maybe?). 

These are the rest of the OIDC-related properties, let me know if you need more info:
```text
quarkus.oidc.enabled=${OIDC_ENABLE:true}
azure.tenant-id=...
quarkus.oidc.auth-server-url=https://login.microsoftonline.com/${azure.tenant-id}/v2.0
quarkus.oidc.application-type=web_app
quarkus.oidc.client-id=...
quarkus.oidc.credentials.secret=...
quarkus.oidc.authentication.scopes=offline_access
quarkus.oidc.token-state-manager.split-tokens=true
quarkus.oidc.token.refresh-expired=true
quarkus.oidc.token.refresh-token-time-skew=5S
quarkus.oidc.roles.role-claim-path=roles
```

### Expected behavior

Authentication proceeds OK.

### Output of `uname -a` or `ver`

Darwin C8049 20.5.0 Darwin Kernel Version 20.5.0: Sat May  8 05:10:33 PDT 2021; root:xnu-7195.121.3~9/RELEASE_X86_64 x86_64

### Output of `java -version`

openjdk version "16.0.1" 2021-04-20 OpenJDK Runtime Environment (build 16.0.1+9-24) OpenJDK 64-Bit Server VM (build 16.0.1+9-24, mixed mode, sharing)

### Quarkus version or git rev

2.3.1.Final

**Comment by sberyozkin** [comment]
@artkonr Hi, thanks for reporting it, it will be fixed in #21150. `2.4.1.Final` release process has started so this fix may not make it - but for now please use a deprecated property - it will only be removed in `2.5.0.CR1` if #21150 makes it to the `2.4.x` line (`2.4.1` or `2.4.2`)

**Comment by artkonr** [comment]
@sberyozkin, gotcha, thanks for taking care of it!
